app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
```

Comment:
Please solve the n+1 problem when fetching all posts and their comments for a user in the `posts#index`. For example:
@posts = @user.posts.includes(:comments)

Reply:
Thank you for pointing that out.